mlbaklava/commands.py

```diff
@@ -114,20 +115,96 @@ def run(self):
             entrypoint=entry,
             requirements=requirements,
             python_version=python_version,
-            dockerlines=dockerlines
+            dockerlines=dockerlines,
+            provider=self.provider
+        )
+
+        self.build(artifacts, entry, directory)
+
+class Process(Docker):
+    description = 'create a docker image for local processing'
+    user_options = [
+
+        # Distribution Configs
+        ('entrypoint=', 'e', "The project entrypoint"),
+        ('provider=', 'p', "The provider this container should be built for"),
+
+        # Docker Configs
+        ('idfile=', 'f', "Save image name to the given file"),
+        ('tag=', 't', "Tag the image with a string"),
+
+        # Flags
+        ('nobuild', 'b', "Flag to not build the image"),
+        ('hash', 'h', "Silently build and output image hash"),
+    ]
+
+    def initialize_options(self):
+
+        # Distribution Configs
+        self.entrypoint = None
+        self.pyver = None
+        self.provider = None
+
+        # Docker Configs
+        self.idfile = None
+        self.tag = None
+
+        # Flags
+        self.nobuild = False
+        self.hash = False
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+
+        # Get the entrypoint
+        entry = entrypoint.get(
+            entry=self.distribution.entry_points,
+            key='mlbaklava.processing',
+            name=self.entrypoint
+        )
+
+        # Create the source distribution
+        self.run_command('sdist')
+        sdist = self.get_finalized_command('sdist')
+        assert len(sdist.archive_files), RuntimeError(
+            'Expected a single archive to be created from source distribution'
+        )
+
+        # Extract path to the distribution
+        path = sdist.archive_files[0]
+        directory = sdist.dist_dir
+        archive = os.path.relpath(path, directory)
+
+        # Extract distribution configurations
+        requirements = self.distribution.install_requires
+        python_version = self.distribution.python_version
+        dockerlines = self.distribution.dockerlines
+
+        # Build docker distribution files
+        artifacts = distribution.processing(
+            path=directory,
+            archive=archive,
+            entrypoint=entry,
+            requirements=requirements,
+            python_version=python_version,
+            dockerlines=dockerlines,
+            provider=self.provider
         )
 
         self.build(artifacts, entry, directory)
 
 
-class Predict(Docker):
+class Deploy(Docker):
     description = 'create a docker image for local prediction'
     user_options = [
 
         # Distribution Configs
         ('entrypoint=', 'e', "The project entrypoint"),
         ('initializer=', 'n', "The project initializer"),
         ('workers=', 'e', "The number of worker processes host in the image (default=8)"),
+        ('provider=', 'p', "The provider to create this container for"),
```

Comment:
We can also add flag to enable data adapter so that adapter becomes part of native implementation. 

Default value can be false. This would make Adapter optional and easy to implement instead of doing funky error handling